### PR TITLE
[CIRCLE-37092] Article suggester/related resources experiment

### PR DESCRIPTION
**Pull request by rsylvian**
Ticket: [CIRCLE-37092](https://circleci.atlassian.net/browse/CIRCLE-37092)

# Description
- Add a `suggested` [Front Matter](https://jekyllrb.com/docs/front-matter/) variable to store recommendation.
- Set `suggested: false` in the config for all pages so it doesn't show up by mistake.
- Add `relatedResources.js` in the `experiments/` directory to properly handle the experiment.
- Add support for audience attributes in the `OptimizelyClient`
- Pass `windowWidth` as an attribute to disqualify everyone on mobile (Table of content is not shown on mobile)

# Reasons
We are going to run an experiment on the top 5 doc pages for an MVP version of the Article Suggester. This is a fairly simple experiment (nothing is smart/dynamic and selects the suggested articles) because we first want to validate that this is something users want. Pages + Links can be found here: https://docs.google.com/document/d/1Mg5kXWXfM2OstgqMH8ybN92KmTkqsDs-137NvuU-0gM/edit#

# Screenshots

in control:
Screen Shot 2021-09-20 at 12 24 33 PM

in treatment:
Screen Shot 2021-09-20 at 12 21 44 PM

# Details 

**Configuration Reference**

Screen Shot 2021-09-25 at 3 00 21 PM

**Env Vars**

image

**Workflows**

image

**Sample Config**

image

**Local CLI**

image

### CODE REVIEW

File: src-js/experiments/relatedResources.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  // https://app.optimizely.com/v2/projects/16812830475/experiments/20638820544/variations
+  window.OptimizelyClient.getVariationName({
+    experimentKey: 'dd_docs_related_resources_mvp_test',
+    groupExperimentName: 'q3_fy22_docs_disco_experiment_group_test'
+  }).then(variation => {
+    if (variation === "treatment") {
+      $('#related-resources').show();
+      // track clicks on each items
+      $("#related-resources li a").click(function () {
+        window.AnalyticsClient.trackAction('related-rerouces-clicked', {
```
Comment (tshen815) [inline src-js/experiments/relatedResources.js:11]: small typo here to look out for:
```suggestion
        window.AnalyticsClient.trackAction('related-resources-clicked', {
```

Comment (teesloane) [inline src-js/experiments/relatedResources.js:11]:
```suggestion
        window.AnalyticsClient.trackAction('related-resources-clicked', {
```

Comment (teesloane) [reply src-js/experiments/relatedResources.js:11]: Also - maybe preface this with `doc-` or `dd-` ?

**Review by teesloane — APPROVED** [review]
Looks great.

File: src-js/experiments/relatedResources.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+$(() => {
+  // https://app.optimizely.com/v2/projects/16812830475/experiments/20638820544/variations
+  window.OptimizelyClient.getVariationName({
+    experimentKey: 'dd_docs_related_resources_mvp_test',
+    groupExperimentName: 'q3_fy22_docs_disco_experiment_group_test',
+    attributes: {
+      windowWidth: window.innerWidth ?? 0
+    }
```
Comment (tshen815) [inline src-js/experiments/relatedResources.js:8]: Just curious on what this is for?

File: src-js/optimizely.js [hunk 3]
```diff
@@ -40,6 +40,9 @@ class OptimizelyClient {
         return reject({error: "Missing required options"});
       }
 
+      // defines additional attributes we will want to send to optimizely to qualify/disqualify a user
+      const attributes = options.attributes ?? {};
+
```
Comment (tshen815) [inline src-js/optimizely.js:45]: Ah I see, going off of my previous comment, is the attribute that you included in `getVariationName` another check that the user needs to satisfy to be able to qualify for an experiment?

**Review by tshen815 — APPROVED** [review]
Looks great overall!

**Review by nramabad — APPROVED** [review]

**Pull request merged**